db/migrate/20161018134050_create_widgets.rb

```diff
@@ -0,0 +1,23 @@
+class CreateWidgets < ActiveRecord::Migration
+  def change
+    create_table :widgets do |t|
+      t.integer :widget_type, null: false
+      t.integer :position, null: false
+      t.text :properties
```

Comment:
Well we're already married to PostgreSQL anyway, so that's why I don't see why not use its advanced features.